### PR TITLE
automatically configure database_tests config for developers when running tests.

### LINKED ISSUE #6635: Move phpunit.xml.dist to phpunit.xml in PHPUnit bootstrap or when executing tests
Either on each test run or only if it doesn't exist yet. To make this useful we should replace `REMOTE_ADDR`, `HTTP_HOST` and `REQUEST_URI` automatically. I think we have at least the URI in `SettingsPiwik`? Not sure about the rest. If not, maybe we could read them from config? 

Currently, a developer has to prepare the DB test config and adjust the phpunit.xml. Having everything only in one place makes it easier to configure and to explain in documentations / blog posts. 

By copying the XML all the time it also prevents the issue that sometimes the `.dist` files changes and people are not aware that they have to update their local file. I'm not sure if an updated XML file would take effect immediately or with the next test run but either way it would be still better. 

Not sure what the best way is to resolve it but would be nice to have it. Maybe we can even get rid of the `.dist` file at all and define those 3 variables in bootstrap.php?

### PR DESCRIPTION
**Pull request by tsteur**
fixes #6635  

I also replaced the check that does a request against Piwik to check
whether it is installed with `SettingsPiwik::isInstalled()` as discussed
recently

### CODE REVIEW

File: core/Updates/2.11.1-b3.php [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    {
+        if (!Development::isEnabled()) {
+            return;
+        }
+
+        $config  = Config::getInstance();
+        $dbTests = $config->database_tests;
+
+        if ($dbTests['username'] === '@USERNAME@') {
+            $dbTests['username'] = 'root';
```
Comment (diosmosis) [inline core/Updates/2.11.1-b3.php]: Why not set it to Config::getInstance()->database['username']?

File: core/Version.php [hunk 2]
```diff
@@ -20,7 +20,7 @@
      * The current Piwik version.
      * @var string
      */
-    const VERSION = '2.11.1-b2';
+    const VERSION = '2.11.1-b3';
```
Comment (mattab) [inline core/Version.php]: fyi I'll release beta3 now so would need to be beta4

File: core/Updates/2.11.1-b3.php (hunk 1 above)
Comment (tsteur) [inline core/Updates/2.11.1-b3.php]: Basically, I just wanted to have it as before. Before, in global config was the default username `root` which I changed to `@USERNAME@`. By setting it back to root for developers who update I make sure it is the same as before. In theory, developers could use a different username for there regular database in `$dbTests['username']`. I used to have this in the `piwik-dev-environment` where I had one user `piwik` for the normal database that Piwik uses and one user `root` for tests.

**Comment by mattab** [comment]
FYI: this pr needs a rebase

**Comment by tsteur** [comment]
done

**Pull request merged**